### ISSUE TITLE
KT20 Minnow

### ISSUE THREAD

**Issue by larsbrinkhoff** [KS10FPGA/KS10FPGA#18]
Hello,

I saw in alt.sys.pdp10 you asked about the Minnow microcode.  Did you get a machine readable microcode source file and an assembler for it?

**Comment by larsbrinkhoff** [comment]
I was hoping to direct my question to @jfcl.

**Comment by jfcl** [comment]
That was a long time ago.   As pdf listing of the microcode is available from: http://www.bitsavers.org/pdf/dec/pdp10/KT20_Minnow/minnow_uCodeSrc.pdf alont with .   I tried to OCR it, but that failed miserably.  I recall spending weeks and weeks hand editing the OCR'd output to create the source code.   I still have the microcode source.

I noticed someone mentioned micro2 assembler on classiccmp recently.  I suspected that it was a VAX tool but I never found a copy.

I created a partially working micro2 assembler using flex and bison.  I could compile the code - but I could never tell if the microcode that was available was anything more than a rough draft.

I was interested in Minnow because it could support more memory than a KS10 or a KL10 - I think it supported 30-bit addressing or something.  It was also interesting because it could boot from it's own microcode and and had debug in microcode like the J11.

In the end, I think the performance was terrible. It was really a 12-bit or 18-bit micromachine depending on how you looked at it.  I think I documented this in a short paper entitled "The Problem(s) with the Minnow PDP-10 Microarchitecture"

In the end, I dropped it an went on to the KS10 FPGA project. It was just better documented and I knew it would actually work.

I also found a copy of my micro assembler.  It wasn't my finest piece of work - just a tool to create the microcode.

In summary, I have machine readable microcode and an assembler for it - with lots of caveats on both claims.

Rob Doyle.

**Comment by larsbrinkhoff** [comment]
Thanks.  I suggest it would be worthwhile to publish the microcode source, assembler, and the paper.

I'll trade you Superfoonly schematics and microcode. :-)

**Comment by jfcl** [comment]
I've archived the bulk of the Minnow work at:

[https://github.com/KS10FPGA/Minnow](https://github.com/KS10FPGA/Minnow)

**Comment by larsbrinkhoff** [comment]
Thanks!  I think this issue can be closed now.

**Comment by jfcl** [comment]
Closed.
